examples/step-3/doc/intro.dox

```diff
@@ -231,6 +238,105 @@ page. An overview of the most fundamental groups of concepts is also available
 
 
+<h3> Solving the linear system </h3>
+
+For a finite element program, the linear system we end up with here is
+relatively small: The matrix has size $1089 \times 1089$, owing to the
+fact that the mesh we use is $32\times 32$ and so there are $33^2=1089$ vertices
+in the mesh. In many of the
+later tutorial programs, matrix sizes in the range of tens of thousands
+to hundreds of thousands will not be uncommon, and with codes such
+as [ASPECT](https://aspect.geodynamics.org) that build on deal.II, we
+regularly solve problems with more than a hundred million equations (albeit
+using parallel computers). In any case, even for the small system here, the
+matrix is much larger than what one typically encounters in an undergraduate
+or most graduate courses, and so the question arises how we can solve such
+linear systems.
+
+The first method one typically learns for solving linear systems is
+[Gaussian elimination](https://en.wikipedia.org/wiki/Gaussian_elimination).
+The problem with this method is that it requires a number of operations that
+is proportional to $N^3$, where $N$ is the number of equations or unknowns
+in the linear system -- more specifically, the number of operations is
+$\frac 23 N^3$, give or take a few. With $N=1089$, this means that we would
+have to do around $861$ million operations. This is a number that is quite
+feasible and it would take modern processors on the order of 1-10 seconds
+to do this. But it is clear that this isn't going to scale: If we have ten times
+as many equations in the linear system (that is, ten times as many unknowns),
+then it would already take 1000-10,000 seconds or on the order of an hour. Make
+the linear system another ten times larger, and it is clear that we can not solve
+it any more on a single computer.
+
+One can rescue the situation somewhat by realizing that only a relatively
+small number of entries in the matrix are nonzero -- that is, the matrix
+is [sparse](https://en.wikipedia.org/wiki/Sparse_matrix). Variations of
+Gaussian elimination can exploit this, making the process substantially
+faster; we will use one such method -- implemented in the SparseDirectUMFPACK
+class -- in step-29 for the first time, among several others than come after
+that. These variations of Gaussian elimination might get us to problem
+sizes on the order of 100,000 or 200,000, but not all that much beyond that.
+
+Instead, what we will do here is take up an idea from 1952: the
+[Conjugate Gradient method](https://en.wikipedia.org/wiki/Conjugate_gradient_method),
+or in short "CG". CG is an "iterative" solver in that it forms a sequence
+of vectors that *converge* to the exact solution; in fact, after $N$ such iterations
+it finds the exact solution if the matrix is symmetric and positive definite.
+The method was originally developed as another way
+to solve a linear system exactly, like Gaussian elimination, but as such
+it had few advantages and was largely forgotten for a few decades. But,
```

Comment:
If I recall the historic texts correctly, one of the main reasons was indeed the much higher sensitivity to roundoff errors than Gaussian elimination (today I would also mention data locality, but that was not an issue initially) - already for moderately ill-conditioned matrices (like `diag([1,2,3,...,100])` you can really see this happen.

Reply:
Yes, round-off is a concern, but that's something that I believe people knew how to deal with -- with iterative refinement. Of course, round-off was a concern also for direct solver, given that they only had single-precision at the time. Basically, the whole issue of forward and backward stability analysis is a product of having only single precision available.

One could expand on all of this, but in the end, that isn't the place for it. I just wanted to briefly touch on the basic motivation to not use direct solvers in FEM codes, and so saving the historical context for another place might be the place to go :-) I don't think you suggested anything else either.

Reply:
Exactly, I didn't want to suggest anything else here, I just wanted to add some thoughts.